test/e2e/serial/tests/configuration.go

```diff
@@ -693,9 +697,220 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
+			klog.Infof("available CPUs per numa on the node: %d, ", cpuNum)
+
+			cpuResources := strconv.Itoa(int(float64(cpuNum) * cpuResourcePercentage))
+			klog.Infof("CPU resources requested to create a topology affinity error deployment %s", cpuResources)
+
+			By("fetching the matching kubeletconfig for the numaresources-operator cr")
+			var nroOperObj nropv1.NUMAResourcesOperator
+			nroKey := objects.NROObjectKey()
+			err := fxt.Client.Get(ctx, nroKey, &nroOperObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroKey.String())
+
+			mcps, err := nropmcp.GetListByNodeGroupsV1(ctx, fxt.Client, nroOperObj.Spec.NodeGroups)
+			Expect(err).ToNot(HaveOccurred(), "cannot get MCPs associated with NUMAResourcesOperator %q", nroOperObj.Name)
+
+			kcList := &machineconfigv1.KubeletConfigList{}
+			err = fxt.Client.List(ctx, kcList)
+			Expect(err).ToNot(HaveOccurred())
+
+			var targetedKC *machineconfigv1.KubeletConfig
+			for _, mcp := range mcps {
+				for i := 0; i < len(kcList.Items); i++ {
+					kc := &kcList.Items[i]
+					kcMcpSel, err := metav1.LabelSelectorAsSelector(kc.Spec.MachineConfigPoolSelector)
+					Expect(err).ToNot(HaveOccurred())
+
+					if kcMcpSel.Matches(labels.Set(mcp.Labels)) {
+						targetedKC = kc
+					}
+				}
+			}
+			Expect(targetedKC).ToNot(BeNil(), "there should be at least one kubeletconfig.machineconfiguration object")
+
+			kcObj, err := kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+			Expect(err).ToNot(HaveOccurred())
+			initialTopologyManagerPolicy := kcObj.TopologyManagerPolicy
+
+			if initialTopologyManagerPolicy != v1beta1.NoneTopologyManagerPolicy {
+				By("modifying topology manager policy under kubeletconfig to none")
+				Eventually(func(g Gomega) {
+					err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
+					Expect(err).ToNot(HaveOccurred())
+
+					kcObj, err = kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+					Expect(err).ToNot(HaveOccurred())
+
+					// Before restarting the scheduler, we change here the topology manager policy to 'none'.
+					// This ensures that if we later change it to 'single NUMA node' and attempt to create
+					// a deployment that is expected to fail due to TopologyAffinityError it will otherwise fail for failed scheduling.
+					// We expect that failure to occur only when the bug is present.
+					kcObj.TopologyManagerPolicy = v1beta1.NoneTopologyManagerPolicy
+					err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
+					Expect(err).ToNot(HaveOccurred())
+
+					err = fxt.Client.Update(ctx, targetedKC)
+					g.Expect(err).ToNot(HaveOccurred())
+				}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+
+				waitForMcpUpdate(fxt.Client, ctx, mcps)
+			}
+
+			defer func() {
+				err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
```

Comment:
let's use local vars as much as possible in the implemented functions of defer and eventually like err and kcObj, for example the first initialization of each is associated with := or